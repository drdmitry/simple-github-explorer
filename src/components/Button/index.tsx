import type { FC, PropsWithChildren } from "react";

export interface IButtonProps extends PropsWithChildren {
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

export const Button: FC<IButtonProps> = ({ disabled = false, onClick, children }) => {
  return (
    <button
      type="button"
      className="bg-sky-500 bg-gradient-to-br from-sky-300 to-sky-500 hover:to-sky-600 active:to-sky-700 text-white shadow-sm font-bold py-2 px-4 rounded-md focus:outline-none focus:shadow-outline disabled:from-sky-100 disabled:to-sky-200 disabled:cursor-no-drop"
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

export default Button;
