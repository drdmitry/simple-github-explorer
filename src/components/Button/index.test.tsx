import React from "react";
import { render } from "@testing-library/react";

import Button from "./";

describe("Button", () => {
  test("renders Button", () => {
    // Render the Button component
    const { getByText, container } = render(
      <Button>Button Content</Button>
    );

    // Assert that the button element exists and has the correct position
    const buttonElement = container.querySelector("button");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveAttribute("type", "button");

    // Assert that the Button Content is rendered
    const contentElement = getByText("Button Content");
    expect(contentElement).toBeInTheDocument();
  });

  test("renders disabled Button", () => {
    // Render the Button component
    const { container } = render(
      <Button disabled>Button Content</Button>
    );

    // Assert that the button element is disabled
    const buttonElement = container.querySelector("button");
    expect(buttonElement).toHaveAttribute('disabled');
  });

  test("renders Button unchanged", () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
});
