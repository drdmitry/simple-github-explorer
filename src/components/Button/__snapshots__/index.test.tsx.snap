// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button unchanged 1`] = `
<div>
  <button
    class="bg-sky-500 bg-gradient-to-br from-sky-300 to-sky-500 hover:to-sky-600 active:to-sky-700 text-white shadow-sm font-bold py-2 px-4 rounded-md focus:outline-none focus:shadow-outline disabled:from-sky-100 disabled:to-sky-200 disabled:cursor-no-drop"
    type="button"
  />
</div>
`;
